### ISSUE TITLE
Accessibility issues

### ISSUE THREAD

**Issue by biou** [goblindegook/littlefoot#375]
Hi, thanks for this very cool library! I have done some checks on its accessibility and I have found some issues:

- some html attributes are invalid, for example rel="footnote" according to the html5 spec https://www.w3.org/TR/html52/links.html#allowed-keywords-and-their-meanings or alt on aside.
- some duplicate id are generated https://dequeuniversity.com/rules/axe/3.0/duplicate-id
- it seems that the aria design pattern disclosure has been implemented with aria-expanded and aria-controls. https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure But in this pattern, aria-controls is optional, and in our case, the referenced element is most of the time not in the dom, when the popover is not displayed. So I would recommend not having aria-controls here. Moreover, aria-live is not used in this design pattern, so it can also be removed.
- there are some problems with the keyboard navigation. There is a tabindex on a non-interactive element which should be avoided (see https://webaim.org/techniques/keyboard/tabindex in the "important" box), and moreover, the content of the  popover is inserted in the natural tab order, so tabindex is not needed here. When closing the popover, the focus should be again on the button that opened it. The focus is currently lost. (see: https://codepen.io/mfairchild365/pen/RwrKxdV )

**Comment by goblindegook** [comment]
Changes look good, I'd just ask you to update the unit tests and documentation to match them if you can, please.  Thanks again!

**Comment by goblindegook** [comment]
That `tabindex` probably exists to allow the user to scroll the popover on long content. Do you happen to know the appropriate strategy for dealing with overflowing content?

**Comment by biou** [comment]
> That `tabindex` probably exists to allow the user to scroll the popover on long content. Do you happen to know the appropriate strategy for dealing with overflowing content?

Good question, I will need to double check. I'll keep you posted.

**Comment by goblindegook** [comment]
I'll keep the issue open. In the meantime, I merged your changes, thanks for taking the time to make them.

**Comment by biou** [comment]
You are right about the tabindex=0, it should be kept because of the scrollable region https://dequeuniversity.com/rules/axe/3.5/scrollable-region-focusable I will submit a pull request soon to put it back.

**Comment by goblindegook** [comment]
Would it be better if it was done conditionally? We can set `tabindex="0"` on the `content` element when it scrolls, and remove it when it doesn't, here:

https://github.com/goblindegook/littlefoot/blob/e5d7e45e492572a936b6240423a5da3138b87d2c/src/dom/footnote.ts#L87

**Comment by goblindegook** [comment]
Pushed https://github.com/goblindegook/littlefoot/commit/0ecde148758d013d01d9e2ac415285a57574fa64 to add `tabindex="0"` for scrollable content.

**Comment by biou** [comment]
sorry, I did not have time to check, but this should be okay.

**Comment by goblindegook** [comment]
No worries. If you spot any more issues, please let me know. You've been great help, thanks again!
